### PR TITLE
[JetNews] Adds Robolectric tests

### PR DESCRIPTION
**Pull request by JoseAlcerreca**

### CODE REVIEW

File: .github/workflows/JetNews.yaml [hunk 1]
```diff
@@ -45,6 +45,10 @@ jobs:
         working-directory: ${{ env.SAMPLE_PATH }}
         run: ./gradlew spotlessCheck assembleDebug lintDebug --stacktrace
 
+      - name: Run local tests
+        working-directory: ${{ env.SAMPLE_PATH }}
+        run: ./gradlew testDebug
```
Comment (manuelvicnt) [inline .github/workflows/JetNews.yaml:50]: Add path? `path: ${{ env.SAMPLE_PATH }}/app/build/reports/tests`

Comment (JoseAlcerreca) [reply .github/workflows/JetNews.yaml:50]: What for? the working directory is set and reports are collected in another task

File: .github/workflows/JetNews.yaml [hunk 2]
```diff
@@ -45,6 +45,10 @@ jobs:
         working-directory: ${{ env.SAMPLE_PATH }}
         run: ./gradlew spotlessCheck assembleDebug lintDebug --stacktrace
 
+      - name: Run local tests
```
Comment (chrisbanes) [inline .github/workflows/JetNews.yaml:48]: Run the `testDebug` task in the `gradlew` task above? That lets Gradle schedule it as appropriate.

File: JetNews/app/build.gradle [hunk 3]
```diff
@@ -111,6 +128,10 @@ dependencies {
     androidTestImplementation "androidx.compose.ui:ui-test-junit4:$compose_version"
 
     debugImplementation "androidx.compose.ui:ui-test-manifest:$compose_version"
+
+    // Robolectric dependencies
+    testImplementation "androidx.compose.ui:ui-test-junit4:$compose_version"
+    testImplementation("org.robolectric:robolectric:4.5.1")
```
Comment (chrisbanes) [inline JetNews/app/build.gradle:134]: 4.6.1?

Comment (chrisbanes) [reply JetNews/app/build.gradle:134]:
```suggestion
    testImplementation "org.robolectric:robolectric:4.5.1"
```

File: JetNews/app/src/test/resources/robolectric.properties [hunk 4]
```diff
@@ -0,0 +1 @@
+qualifiers=w360dp-h640dp-xhdpi
```
Comment (chrisbanes) [inline JetNews/app/src/test/resources/robolectric.properties:1]: Add a comment

File: .github/workflows/JetNews.yaml (hunk 2 above)
Comment (JoseAlcerreca) [reply .github/workflows/JetNews.yaml:48]: I don't know what that means but sure

File: .github/workflows/JetNews.yaml [hunk 5]
```diff
@@ -41,9 +41,9 @@ jobs:
             ~/.gradle/caches/build-cache-*
           key: gradle-${{ hashFiles('checksum.txt') }}
 
-      - name: Build project
+      - name: Build project and run local tests
         working-directory: ${{ env.SAMPLE_PATH }}
-        run: ./gradlew spotlessCheck assembleDebug lintDebug --stacktrace
+        run: ./gradlew spotlessCheck assembleDebug lintDebug testDebug --stacktrace
```
Comment (chrisbanes) [inline .github/workflows/JetNews.yaml:46]: This could just be `./gradlew build`?

Comment (JoseAlcerreca) [reply .github/workflows/JetNews.yaml:46]: build runs all the release stuff, however buildDebug does not run local tests.

**Review by chrisbanes — APPROVED** [review]

**Pull request merged**